solidity/random-beacon/test/RandomBeacon.Authorization.test.ts

```diff
@@ -710,26 +716,219 @@ describe("RandomBeacon - Authorization", () => {
-            ["requestAuthorizationDecrease(address,address,uint96)"](
-              stakingProvider.address,
-              randomBeacon.address,
-              deauthorizingSecond
-            )
         })
 
         after(async () => {
           await restoreSnapshot()
         })
 
-        it("should overwrite the previous request", async () => {
-          expect(
-            await randomBeacon.pendingAuthorizationDecrease(
-              stakingProvider.address
+        context("when change period is equal delay", () => {
+          before(async () => {
+            // this should be the default situation from the fixture setup so we
+            // just confirm it here
+            const {
+              authorizationDecreaseDelay,
+              authorizationDecreaseChangePeriod,
+            } = await randomBeacon.authorizationParameters()
+            expect(authorizationDecreaseDelay).to.equal(
+              authorizationDecreaseChangePeriod
             )
-          ).to.be.equal(deauthorizingSecond)
+          })
+
+          context("when delay passed", () => {
+            before(async () => {
+              await createSnapshot()
+              await helpers.time.increaseTime(params.authorizationDecreaseDelay)
+
+              await staking
+                .connect(authorizer)
+                ["requestAuthorizationDecrease(address,address,uint96)"](
+                  stakingProvider.address,
+                  randomBeacon.address,
+                  deauthorizingSecond
+                )
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            it("should overwrite the previous request", async () => {
+              expect(
+                await randomBeacon.pendingAuthorizationDecrease(
+                  stakingProvider.address
+                )
+              ).to.be.equal(deauthorizingSecond)
+            })
+          })
+
+          context("when delay did not pass", () => {
+            before(async () => {
+              await createSnapshot()
+              await helpers.time.increaseTime(
+                params.authorizationDecreaseDelay - 60 // -1min
+              )
+
+              await staking
+                .connect(authorizer)
+                ["requestAuthorizationDecrease(address,address,uint96)"](
+                  stakingProvider.address,
+                  randomBeacon.address,
+                  deauthorizingSecond
+                )
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
+
+            it("should overwrite the previous request", async () => {
+              expect(
+                await randomBeacon.pendingAuthorizationDecrease(
+                  stakingProvider.address
+                )
+              ).to.be.equal(deauthorizingSecond)
+            })
+          })
+        })
+
+        context("when change period is zero", () => {
```

Comment:
Can we have two test cases under `"when change period is zero"`:
1. `requestAuthorizationDecrease` is called right away without any time increase
1. `requestAuthorizationDecrease` is called after the delay period